src/cloud-api-adaptor/podvm/hack/cross-build-extras.sh

```diff
@@ -17,5 +17,12 @@ libc=$([[ $ARCH =~ s390x ]] && echo "gnu" || echo "musl")
 rustTarget="$ARCH-unknown-linux-$libc"
 
 rustup target add "$rustTarget"
-apt install -y "qemu-system-$ARCH"
-apt install -y "gcc-$ARCH-linux-$libc"
+
+source /etc/os-release || source /usr/lib/os-release
+if [[ ${ID_LIKE:-} == *"debian"* ]]; then
+    apt install -y "qemu-system-$ARCH"
+    apt install -y "gcc-$ARCH-linux-$libc"
+elif [[ "${ID_LIKE:-}" =~ "fedora" ]]; then
```

Comment:
In my workstation with Fedora 40 there isn't `ID_LIKE`:
```
wmoschet@wmoschet-thinkpadp16vgen1:/tmp$ grep ID_LIKE /etc/os-release
wmoschet@wmoschet-thinkpadp16vgen1:/tmp$ grep ^ID /etc/os-release
ID=fedora
```

Also, maybe take that opportunity to add an `else echo "Unsupported distro $ID"; exit 1`

Reply:
Ah, I was expected that like ubuntu & debian share the same ID_LIKE, that rhel & fedora would as well as it looks like RHEL follows that pattern:
```
# cat /etc/os-release 
NAME="Red Hat Enterprise Linux"
VERSION="8.6 (Ootpa)"
ID="rhel"
ID_LIKE="fedora"
```

I've updated it to handle ID=fedora and add the unsupported distro. Thanks!